src/repositories/jobRepository.ts

```diff
@@ -93,6 +93,29 @@ export class JobRepository extends BaseRepository {
     await this._queueConnector.sendMessage(new JobQueueMessage(jobId, status, 0, taskId), url, delay);
   }
 
+  async findOneAndUpdateExecutionTime(id: string, setValues: { [key: string]: number }): Promise<Job | null> {
+    const query = {
+      _id: new objectId(id),
+    };
+    const update = { $set: setValues };
+    const options = { sort: { priority: -1, createdTime: 1 }, returnNewDocument: true };
+    const response = await this.findOneAndUpdate(
+      query,
+      update,
+      options,
+      `Mongo Timeout Error: Timed out while retrieving job`
+    );
+
+    if (!response) {
+      throw new InvalidJobError('JobRepository:getOneQueuedJobAndUpdate unable to update job execution time');
```

Comment:
Question: will this end the build? Do we want an error on saving execution time to be a build-ending error?

Reply:
@mmeigs good question, I believe this will end the build, which seems to handle it [here](https://github.com/mongodb/docs-worker-pool/blob/d14e3a702bdd4436369389716bb6ed3be4be36af/src/job/jobHandler.ts#L495-L496). That is a good question. I can ask the channel but, I wouldn't see a reason for this to be a build-ending error.

Reply:
Great! Yeah, I think we're all on the same page with this. So, this could just be an error log rather than a throw.